### PR TITLE
docs(DEPRECATION.md): document removal of `contextToDataSourceId`

### PR DESCRIPTION
**Pull request by hassankhan**
**Check List**
- [x] Tests has been run in packages where changes made if available
- [x] Linter has been run for changed code
- [x] Tests for the changes have been added if not covered yet
- [x] Docs have been added / updated if required

**Description of Changes Made (if issue reference is not provided)**

Document removal of `contextToDataSourceId`.

### CODE REVIEW

**Review by igorlukanin — APPROVED** [review]
LGTM 👍

@hassankhan, I wonder if we'll need a third column ("removed") for the actual date and renaming the second one to "to be removed" once we actually start removing things.

File: DEPRECATION.md [hunk 1]
```diff
@@ -33,12 +33,27 @@ features:
   migrate to alternatives. In such cases, a warning may be printed, and users
   should not rely on this feature.
 
-| Status     | Feature                                                               | Deprecated | Remove    |
-| ---------- | --------------------------------------------------------------------- | ---------- | --------- |
-| Deprecated | [Embedding Cube.js within Express](#embedding-cubejs-within-express)  | v0.24.0    | June 2021 |
-| Deprecated | [`CUBEJS_ENABLE_TLS`](#cubejs_enable_tls)                             | v0.23.11   | June 2021 |
-| Deprecated | [`hearBeatInterval`](#hearbeatinterval)                               | v0.23.8    | June 2021 |
-| Deprecated | [Node.js 8](#nodejs-8)                                                | v0.22.4    |   v0.26.0 |
+| Status     | Feature                                                              | Deprecated | Remove       |
```
Comment (rpaik) [inline DEPRECATION.md:36]:
```suggestion
| Status     | Feature                                                              | Deprecated | Removed/To be Removed       |
```

Comment (rpaik) [reply DEPRECATION.md:36]: Is this too wordy?

**Comment by hassankhan** [comment]
@igorlukanin @rpaik I cribbed the format from other projects on GitHub (notably Docker) which had a single 'Remove' column for both past and future removal dates. It's always something we could do as a separate PR though 👍

**Review by rpaik — APPROVED** [review]
